src/menu/Menu.ts

```diff
@@ -0,0 +1,311 @@
+import { assign } from '@dojo/core/lang';
+import { createTimer } from '@dojo/core/util';
+import uuid from '@dojo/core/uuid';
+import { Handle } from '@dojo/interfaces/core';
+import { v, w } from '@dojo/widget-core/d';
+import { DNode } from '@dojo/widget-core/interfaces';
+import ThemeableMixin, { theme, ThemeableProperties } from '@dojo/widget-core/mixins/Themeable';
+import WidgetBase from '@dojo/widget-core/WidgetBase';
+import MenuItem, { MenuItemProperties } from './MenuItem';
+import * as css from './styles/menu.css';
+
+export type Role = 'menu' | 'menubar';
+
+export interface MenuProperties extends ThemeableProperties {
+	/**
+	 * Only applies to nested menus. Either a flag indicating whether the widget instance should handle animating
+	 * between the visible and hidden states, or a function that manually handles the animation. If true (the default),
+	 * then the menu will slide into and out of view like an accordion. If false, then any animation should be handled
+	 * in the CSS, as the menu will just snap open/shut.
+	 */
+	animate?: boolean | ((element: HTMLElement, hidden: boolean) => void);
+
+	/**
+	 * Indicates whether the menu is disabled. If true, then the widget will ignore events.
+	 */
+	disabled?: boolean;
+
+	/**
+	 * Indicates whether the menu should be displayed/hidden via a click event. If false, then the menu is toggled
+	 * via a hover event. Defaults to true.
+	 */
+	expandOnClick?: boolean;
+
+	/**
+	 * Only applies to menus toggled into view with a hover event. The amount of time (in milliseconds) after a
+	 * mouseleave event that should pass before the menu is hidden. Defaults to 300ms.
+	 */
+	hideDelay?: number;
+
+	/**
+	 * Whether the menu is hidden. Defaults to true if a label is specified (i.e., the menu is controlled by a
+	 * a link); false otherwise.
+	 */
+	hidden?: boolean;
+
+	/**
+	 * The widget ID. Defaults to a random string.
+	 */
+	id?: string;
+
+	/**
+	 * The text or properties for a MenuItem widget that is used to control the menu.
+	 */
+	label?: string | MenuItemProperties;
+
+	/**
+	 * Indicates whether the menu is nested within another menu. Useful for styling, this does not affect
+	 * functionality. Defaults to false.
+	 */
+	nested?: boolean;
```

Comment:
Strictly speaking, we could use child selectors to manage the same thing. But since `composes` only works with single selectors, I added the extra class to be safe.